x-pack/test/api_integration/apis/ml/results/get_partition_fields_values.ts

```diff
@@ -0,0 +1,247 @@
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+
+  function getJobConfig(jobId: string, enableModelPlot = true) {
+    return {
+      job_id: jobId,
+      description:
+        'mean/min/max(responsetime) partition=airline on farequote dataset with 1h bucket span',
+      groups: ['farequote', 'automated', 'multi-metric'],
+      analysis_config: {
+        bucket_span: '1h',
+        influencers: ['airline'],
+        detectors: [
+          { function: 'mean', field_name: 'responsetime', partition_field_name: 'airline' },
+          { function: 'min', field_name: 'responsetime', partition_field_name: 'airline' },
+          { function: 'max', field_name: 'responsetime', partition_field_name: 'airline' },
+        ],
+      },
+      data_description: { time_field: '@timestamp' },
+      analysis_limits: { model_memory_limit: '20mb' },
+      model_plot_config: { enabled: enableModelPlot },
+    } as Job;
+  }
+
+  function getDatafeedConfig(jobId: string) {
+    return {
+      datafeed_id: `datafeed-${jobId}`,
+      indices: ['ft_farequote'],
+      job_id: jobId,
+      query: { bool: { must: [{ match_all: {} }] } },
+    } as Datafeed;
+  }
+
+  async function createMockJobs() {
+    await ml.api.createAndRunAnomalyDetectionLookbackJob(
+      getJobConfig('fq_multi_1_ae'),
+      getDatafeedConfig('fq_multi_1_ae')
+    );
+
+    await ml.api.createAndRunAnomalyDetectionLookbackJob(
+      getJobConfig('fq_multi_2_ae', false),
+      getDatafeedConfig('fq_multi_2_ae')
+    );
+  }
+
+  describe('GetAnomaliesTableData', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/farequote');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await createMockJobs();
+    });
+
+    after(async () => {
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should fetch anomalous only field values within the time range with an empty search term sorting by anomaly score', async () => {
+      const requestBody = {
+        jobId: 'fq_multi_1_ae',
+        criteriaFields: [{ fieldName: 'detector_index', fieldValue: 0 }],
+        earliestMs: 1454889600000, // February 8, 2016 12:00:00 AM GMT
+        latestMs: 1454976000000, // February 9, 2016 12:00:00 AM GMT,
+        searchTerm: {},
+        fieldsConfig: {
+          partition_field: {
+            applyTimeRange: true,
+            anomalousOnly: true,
+            sort: { by: 'anomaly_score', order: 'desc' },
+          },
+        },
+      };
+
+      const { body, status } = await supertest
+        .post(`/api/ml/results/partition_fields_values`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .set(COMMON_REQUEST_HEADERS)
+        .send(requestBody);
+      ml.api.assertResponseStatusCode(200, status, body);
```

Comment:
We run this bit a couple times, so might be worth an extra function in this file?

Reply:
refactored in https://github.com/elastic/kibana/pull/133519/commits/d878e0e162e9b1b2d3dddbe627962bd25174b31b